solr/core/src/java/org/apache/solr/update/SolrIndexConfig.java

```diff
@@ -303,6 +304,10 @@ private MergePolicy buildMergePolicy(SolrResourceLoader resourceLoader, IndexSch
         new Class[] { SolrResourceLoader.class, MergePolicyFactoryArgs.class, IndexSchema.class },
         new Object[] {resourceLoader, mpfArgs, schema });
 
+    if (mpf instanceof UninvertDocValuesMergePolicyFactory) {
```

Comment:
When Solr loads a deprecated class (see SolrResourceLoader line 537), it detects this and logs a warning. I think that's fine.

Reply:
Ah, good point! I'd started with the warning here and added the deprecation as an afterthought but yes it's much cleaner to have only the regular deprecation logging, thanks!